editor/project_export.cpp

```diff
@@ -724,6 +724,11 @@ void ProjectExportDialog::_export_project() {
 
 	export_project->set_access(FileDialog::ACCESS_FILESYSTEM);
 	export_project->clear_filters();
+
+	if (default_filename == "") {
+		default_filename = ProjectSettings::get_singleton()->get("application/config/name");
```

Comment:
This should be done in the constructor instead where `default_filename` is initialized (as a `String()` if not configured otherwise): https://github.com/flashyincceo/godot/blob/e219fcb06d3eb7c99ed308425ef5c68d61a55c03/editor/project_export.cpp#L993